geopandas/tools/sjoin.py

```diff
@@ -196,27 +175,21 @@ def _geom_predicate_query(left_df, right_df, predicate, distance):
         DataFrame with matching indices in
         columns named `_key_left` and `_key_right`.
     """
-    with warnings.catch_warnings():
-        # We don't need to show our own warning here
-        # TODO remove this once the deprecation has been enforced
-        warnings.filterwarnings(
-            "ignore", "Generated spatial index is empty", FutureWarning
```

Comment:
What was this warning about? (that's for a warning we already removed in the meantime?)

Reply:
Yep, https://github.com/geopandas/geopandas/pull/1444/ in geopandas/base.py::_generate_sindex